serverless/packages/functions/src/trainspace/get_all_trainspace.ts

```diff
@@ -0,0 +1,39 @@
+import { APIGatewayProxyHandlerV2 } from "aws-lambda";
+import parseJwt from "@dlp-sst-app/core/parseJwt";
+import { DynamoDBClient, QueryCommand } from '@aws-sdk/client-dynamodb';
+import { DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';
+
+export const handler: APIGatewayProxyHandlerV2 = async (event) => {
+    if (event) {
+        const user_id: string = parseJwt(event.headers.authorization ?? "")[
+            "user_id"
+        ];
+
+        const client: DynamoDBClient = new DynamoDBClient({});
+        const docClient: DynamoDBDocumentClient = DynamoDBDocumentClient.from(client);
+        
+        const getCommand = new QueryCommand({
+            TableName: "trainspace",
+            IndexName: "uid",
+            KeyConditionExpression: "uid = :uid",
+            ExpressionAttributeValues: {
+                ":uid" : 
+                { 
+                    S: user_id
+                }
+            }
+        });
+
+        const results = await docClient.send(getCommand);

```

Comment:
From my understanding Dynamo DB will paginate responses when a Get/query operation is called. We might have to use LastEvaluatedKey to get all entries/trainspaces corresponding to a given user id. I think we should explore how to use LastEvaluatedKey in order to ensure the get all trainspace endpoint works correctly

https://w11i.me/paging-dynamodb-queries-with-typescript-generators

https://stackoverflow.com/questions/44589967/how-to-fetch-scan-all-items-from-aws-dynamodb-using-node-js

@alantao912

Reply:
I will look into this. Thanks for bringing it to my attention.